File: 0101-symmetric-tree/0101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True
        return self.isSame(root.left, root.right)
    
    def isSame(self, leftroot, rightroot):
        #If the left & right node are both null, it will return True
        if leftroot == None and rightroot == None:
            return True
        #If only one of the nodes is equal to null, return False
        if leftroot == None or rightroot == None:
            return False
        if leftroot.val != rightroot.val:
            return False
        
        return self.isSame(leftroot.right, rightroot.left) and self.isSame(leftroot.left, rightroot.right)
        

File: 0101-symmetric-tree/0101-symmetric-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {    
    let isSame = function(leftRoot, rightRoot) {
        if (!leftRoot && !rightRoot) {
            return true;
        }
        if (!leftRoot || !rightRoot) {
            return false;
        }
        if(leftRoot.val !== rightRoot.val) {
            return false;
        }
        return isSame(leftRoot.left, rightRoot.right) && isSame(leftRoot.right, rightRoot.left);
    }
    if(!root) {
        return true;
    }
    return isSame(root.left, root.right);    
};